src/runtime/StateSerialization/MaybeMemberInfo.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.IO;
+
+namespace Python.Runtime
+{
+     [Serializable]
+    internal struct MaybeMemberInfo<T> : ISerializable where T: MemberInfo
+    {
+        public static implicit operator MaybeMemberInfo<T> (T ob) => new MaybeMemberInfo<T>(ob);
+
+        // .ToString() of the serialized object
+        const string SerializationName = "s";
+        // The ReflectedType of the object
+        const string SerializationType = "t";
+        const string SerializationFieldName = "f";
+        string name;
+        MemberInfo info;
+
+        [NonSerialized]
+        Exception deserializationException;
+
+        public string DeletedMessage 
+        {
+            get
+            {
+                return $"The .NET {typeof(T)} {name} no longer exists. Cause: " + deserializationException?.Message ;
+            }
+        }
+
+        public T Value
+        {
+            get
+            {
+                if (info == null)
+                {
+                    throw new SerializationException(DeletedMessage, innerException: deserializationException);
+                }
+                return (T)info;
+            }
+        }
+
+        public string Name {get{return name;}}
+        public bool Valid => info != null;
+
+        public override string ToString()
+        {
+            return (info != null ? info.ToString() : $"missing type: {name}");
+        }
+
+        public MaybeMemberInfo(T fi)
+        {
+            info = fi;
+            name = info?.ToString();
+            deserializationException = null;
+        }
+
+        internal MaybeMemberInfo(SerializationInfo serializationInfo, StreamingContext context)
+        {
+            // Assumption: name is always stored in "s"
+            name = serializationInfo.GetString(SerializationName);
+            info = null;
+            deserializationException = null;
+            try
+            {
+                var tp = Type.GetType(serializationInfo.GetString(SerializationType));
+                if (tp != null)
```

Comment:
Is it reasonable to expect a null type? Otherwise, maybe just assume it's non-null and let the exception get thrown.

Reply:
It is. If the type is not found, the method returns null. I added it because it doesn't really makes sense to have a NullReferenceException as the exception for this.